java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -2552,6 +2561,10 @@ private static ColumnVector from(DType type, Object dataArray) {
         case FLOAT64:
           ret = ColumnVector.fromBoxedDoubles((Double[]) dataArray);
           break;
+        case DECIMAL32:
+        case DECIMAL64:
+          ret = ColumnVector.fromDecimals((BigDecimal[]) dataArray);
```

Comment:
fromDecimals is a far from ideal API because it throws out the actual type we want.  We are also auto-detecting the scale etc.  This is *not* the right API to call for this.